framework/src/eventloop/eventloop_event.c

```diff
@@ -0,0 +1,377 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+ /****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <debug.h>
+#include <signal.h>
+#include <queue.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <libtuv/uv.h>
+#include <libtuv/uv__types.h>
+#include <eventloop/eventloop.h>
+
+#include "eventloop_internal.h"
+
+struct event_group_s {
+	struct event_group_s *flink;
+	int type;
+	sq_queue_t event_list;
+};
+typedef struct event_group_s event_group_t;
+
+struct event_node_s {
+	struct event_node_s *flink;
+	el_event_t *event;
+};
+typedef struct event_node_s event_node_t;
+
+struct event_cb_s {
+	event_callback func;
+	void *cb_data;
+	void *event_data;
+};
+typedef struct event_cb_s event_cb_t;
+
+struct event_data_s {
+	int type;
+	int pid;
+	int event_in;
+	event_cb_t cb_info;
+};
+typedef struct event_data_s event_data_t;

```

Comment:
Could you comment the relation of each type, event_group_t, event_node_t, el_event_t, event_cb_t and event_data_t?
This file has too many types without any description. It is hardly difficult to know what it is.